core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java

```diff
@@ -332,10 +332,16 @@ public static class ExternalCompactionColumnFamily {
       public static final Text NAME = new Text(STR_NAME);
     }
 
-    public static class OnDemandAssignmentStateColumnFamily {
-      public static final String STR_NAME = "ondemand";
+    public static class HostingGoalColumnFamily {
+      public static final String STR_NAME = "hosting_goal";
       public static final Text NAME = new Text(STR_NAME);
+      public static final String ALWAYS = "always"; // tablet should always be hosted
+      public static final String DEFAULT = "default"; // tablet should be unassigned
+      public static final String ONDEMAND = "ondemand"; // ondemand request has been made to assign
+                                                        // tablet
+      public static final String NEVER = "never"; // tablet should never be hosted
```

Comment:
```suggestion
      public static final String ALWAYS = TabletHostingGoal.ALWAYS.name(); // tablet should always be hosted
      public static final String DEFAULT = TabletHostingGoal.DEFAULT.name(); // tablet should be unassigned
      public static final String ONDEMAND = TabletHostingGoal.ONDEMAND.name(); // ondemand request has been made to assign
                                                        // tablet
      public static final String NEVER = TabletHostingGoal.NEVER.name(); // tablet should never be hosted
```

Reply:
Updated based on your suggestion in d579d7b